.github/workflows/main.yml

```diff
@@ -15,4 +15,37 @@ jobs:
       - name: Build
         run: docker build -t aspera-client .
       - name: Test
-        run: docker run --rm aspera-client idr0001 md5sum.txt /tmp/idr0001/
\ No newline at end of file
+        run: docker run --rm aspera-client idr0001 md5sum.txt /tmp/idr0001/
+  upload:
+    needs: build
+    if: startsWith(github.ref, 'refs/tags')
```

Comment:
Couldn't several of these steps be run for PRs?

Reply:
With [3ea6e7b](https://github.com/IDR/aspera-client-docker/pull/4/commits/3ea6e7b92af203861299ffa3285f97db1ea1221f) and [919b155](https://github.com/IDR/aspera-client-docker/pull/4/commits/919b1558598dcbd06962a388217cfcd74e0b271b), the logic should now be tested on every run with the exception of:
1. the login step (since secrets are not available to Pull Requests)
2. the push which only happens on tag.

I agree running as many of these steps in dry-run mode should allow to prevent early failures. Should we get this merged and tagged to confirm the push logic still works as expected? We might want to review other instances of Docker upload workflows to propagate similar changes /cc @jburel